### PR TITLE
Added operator_allowed! for Timestamp types

### PR DESCRIPTION
**Pull request by csharad**
For issue #1056. This is my first PR ever. Hope it works.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,5 @@ target
 Cargo.lock
 !diesel_cli/Cargo.lock
 .env
+.idea/
+*.iml
```
Comment (killercup) [inline .gitignore]: Can you instead add those to you global gitignore file?

**Comment by Eijebong** [comment]
I think that error is because we're manually allowing `Add` for this column in the test suite. The line needs to be removed.

**Comment by csharad** [comment]
Only `Timestamp` and `Date` are `Add` and `Sub`. Should I remove others from `__diesel_generate_ops_impls_if_date_time` or add impls for them?

**Comment by killercup** [comment]
This LGTM, but

r? @Eijebong

**Comment by Eijebong** [comment]
Why aren't other types `Add` and `Sub` ? I guess it doesn't really matter for this PR since we can easily add them later to the list.

**Comment by killercup** [comment]
Thanks! We should probably add a test for this as well, but can do so in a follow-up PR

**Pull request merged**